test/unit/source/tile.test.js

```diff
@@ -191,12 +192,33 @@ test('expiring tiles', (t) => {
             expires: date.toString()
         });
 
-        // this shouldn't happen, but if both expiry data are set, cacheControl takes precedence
-        t.equal(tile.getExpiry(), tile.timeAdded + 60000, 'cache-control takes precedence over expires');
+        expiryTimeout = tile.getExpiryTimeout();
+        t.ok(expiryTimeout > 598000 && expiryTimeout < 600000, 'expires header set date as expected');
 
-        delete tile.cacheControl;
+        t.end();
+    });
+
+    t.test('exponential backoff handling', (t) => {
+        const tile = new Tile(new TileCoord(1, 1, 1));
+        tile.state = 'loaded';
+        tile.timeAdded = Date.now();
+
+        tile.setExpiryData({
+            cacheControl: 'max-age=10'
+        });
+
+        const expiryTimeout = tile.getExpiryTimeout();
+        t.ok(expiryTimeout >= 8000 && expiryTimeout <= 10000, 'expiry timeout as expected when fresh');
+
+        tile.expiredRequests = 1;
```

Comment:
I think it would be slightly more reliable if we used the public `tile.setExpiryData()` api here instead of directly setting `tile.expiredRequests`... is it feasible to do this (together with either `setTimeout` to get past the expiration time or else maybe by using Sinon to fake the current time)?